src/model/src/ModelTransformers.cpp

```diff
@@ -457,7 +457,11 @@ bool createReducedModel(const Model& fullModel,
 
             Axis rotationAxis_wrt_newLink1 = newLink1_X_oldLink1*oldJointRevolute->getAxis(oldLink1);
 
-            RevoluteJoint* newJointRevolute = new RevoluteJoint(newLink1,newLink2,newLink1_X_newLink2,rotationAxis_wrt_newLink1);
+            RevoluteJoint* newJointRevolute = new RevoluteJoint(*oldJointRevolute);
+
+            newJointRevolute->setAttachedLinks(newLink1,newLink2);
+            newJointRevolute->setRestTransform(newLink1_X_newLink2);
+            newJointRevolute->setAxis(rotationAxis_wrt_newLink1);
```

Comment:
Ad discussed f2f, this need to be modified to be joint agnostic, I will open an issue so this is clear.